src/vs/workbench/contrib/terminalContrib/accessibility/browser/terminal.accessibility.contribution.ts

```diff
@@ -59,9 +60,20 @@ class AccessibleBufferContribution extends DisposableStore implements ITerminalC
 		private readonly _instance: ITerminalInstance,
 		processManager: ITerminalProcessManager,
 		widgetManager: TerminalWidgetManager,
-		@IInstantiationService private readonly _instantiationService: IInstantiationService
+		@IInstantiationService private readonly _instantiationService: IInstantiationService,
+		@IConfigurationService configurationService: IConfigurationService,
+		@IAccessibilityService accessibilityService: IAccessibilityService
 	) {
 		super();
+		this.add(_instance.onDidRunText(() => {
+			const focusAfterRun = configurationService.getValue(TerminalSettingId.FocusAfterRun);
+			const focusTerminal = focusAfterRun === 'terminal' || (focusAfterRun === 'auto' && accessibilityService.isScreenReaderOptimized());
```

Comment:
I think this condition is wrong and should be `!isScreenReaderOptimized` at the end?

Using a switch makes this sort of thing much more readable imo:

```
switch (focusAfterRun) {
  case 'auto':
    ...
  case 'terminal':
    ...
...
}
```